internal/validators/namespace.go

```diff
@@ -34,8 +34,10 @@ type Namespace struct {
 
 // nsRequest defines the aspects of the admission.Request that we care about.
 type nsRequest struct {
-	ns *corev1.Namespace
-	op k8sadm.Operation
+	nnm   string
+	newns *corev1.Namespace
+	oldns *corev1.Namespace
```

Comment:
Rather than splitting the existing `ns` into three variables, I'd recommend adding `oldNS` and _only_ populating it for updates. E.g. even for delete operations, you put the instance in `ns`, not `oldNS`. That preserves a lot more existing code and would make this change smaller and easier to validate (I talk a bit more about this in DecodeRequest, below).